flink-core/src/main/java/org/apache/flink/configuration/PipelineOptions.java

```diff
@@ -282,4 +282,21 @@ public enum VertexDescriptionMode {
                     .withDescription(
                             "Whether name of vertex includes topological index or not. "
                                     + "When it is true, the name will have a prefix of index of the vertex, like '[vertex-0]Source: source'. It is false by default");
+
+    @PublicEvolving
+    public static final ConfigOption<Boolean> ALLOW_UNALIGNED_SOURCE_SPLITS =
```

Comment:
This should be marked as `@Deprecated("Will be removed in future Flink releases")`.

Reply:
Deprecated doesn't allow for a string as argument but I've added an additional comment.